sklearn/preprocessing/_encoders.py

```diff
@@ -304,7 +304,7 @@ def n_values_(self):
         return self._n_values_
 
     def _handle_deprecations(self, X):
-
+        self._n_values = self.n_values
```

Comment:
I would put this under the line below, as this is also an internal version of a keyword to handle the deprecation